### DISCUSSION TITLE
Add test for session flash message for Livewire

### DISCUSSION THREAD

**Discussion by mrpritchett** [livewire/livewire#4741] [3. Feature ideas]
It would be super awesome to be able to test for the existence of flash messages within Livewire like:

`Livewire::test(SomeComponent::class)->call('method')->assertSeeSessionFlash('success', 'some flash message');`

Rather than being forced to test for an unnecessary event emission or other workaround.

**Comment by joshhanley** [comment]
@mrpritchett Livewire has access to Laravel's assertions. Have you tried `assertSessionHas()`? https://laravel.com/docs/9.x/http-tests#assert-session-has

**Reply by mrpritchett** [reply]
Doh! Can't believe I missed that for some reason. That makes more sense than my "solution." Thank you sir!

**Reply by joshhanley** [reply]
No worries! 🙂

**Comment by robsontenorio** [comment]
> It would be super awesome to be able to test for the existence of flash messages within Livewire like:
> 
> `Livewire::test(SomeComponent::class)->call('method')->assertSeeSessionFlash('success', 'some flash message');`
> 
> Rather than being forced to test for an unnecessary event emission or other workaround.

@mrpritchett  do you mind to share how did you arranged your component and your test? I am using `v3` but flash messages does not exists at `assertSessionHas()`. 

``

**Comment by Swennet** [comment]
Bumping this because I just ran into the same issue as @robsontenorio

`assertSessionHas()` cannot see flash messages from Livewire components, unless you use ugly array syntax like:

```
// In Livewire Component
session()->flash('message', 'Success message!');

// In test
Livewire::test(LiveiwreComponent::class)->assertSessionHas('_flash.new.0', 'message');
```

See also these similar posts of people running into the same issue:
- https://github.com/livewire/volt/issues/54
- https://stackoverflow.com/questions/64930563/assert-that-a-laravel-livewire-component-sees-a-flash-message-rather-than-a-redi
- https://laracasts.com/discuss/channels/testing/livewire-unit-test-flash-data-assertion
